src/audio/types.h

```diff
@@ -57,26 +56,49 @@ class ChannelCount {
     static ChannelCount fromLayout(ChannelLayout layout) {
         switch (layout) {
         case ChannelLayout::Mono:
-            return ChannelCount(1);
+            return mono();
         case ChannelLayout::DualMono:
-            return ChannelCount(2);
+            return stereo();
         case ChannelLayout::Stereo:
-            return ChannelCount(2);
+            return stereo();
         }
         DEBUG_ASSERT(!"unreachable code");
     }
 
+    static constexpr ChannelCount mono() {
+        return ChannelCount(static_cast<value_t>(1));
+    }
+
+    static constexpr ChannelCount stereo() {
+        return ChannelCount(static_cast<value_t>(2));
+    }
+
+    static ChannelCount fromInt(int value) {
+        VERIFY_OR_DEBUG_ASSERT(value >= std::numeric_limits<value_t>::min() &&
+                value <= std::numeric_limits<value_t>::max()) {
+            return ChannelCount();
+        }
+        return ChannelCount(static_cast<value_t>(value));
+    }
+
     explicit constexpr ChannelCount(
             value_t value = kValueDefault)
             : m_value(value) {
     }
+
+    // A limits checking c-tor fom int channel used in many
+    // external libraries
+    explicit ChannelCount(int value)
```

Comment:
By using a plumbing function `static value_t valueFromInt(int)` we could avoid this confusing back-and-forth conversion.